src/Database/DataFeed.php

```diff
@@ -211,9 +217,15 @@ protected function processCollection()
             /*
              * Flush the select, add ID and tag
              */
-            $cleanQuery = $cleanQuery->select(Db::raw($keyName." as id"));
-            $cleanQuery = $cleanQuery->addSelect(Db::raw("(SELECT '".$tag."') as ".$this->tagVar));
-            $cleanQuery = $cleanQuery->addSelect(Db::raw("(SELECT ".$sorting.") as ".$this->sortVar));
+
+            $cast = $this->getDbDialect() === 'pgsql' ? 'text' : 'binary';
+            $idSelect = sprintf("%s as id", $keyName);
+            $tagSelect = sprintf("cast('%s' as %s) as %s", $tag, $cast, $this->tagVar);
```

Comment:
@dimensi I'd prefer this be wrapped in a condition, and the `CAST` call only be used for PostgreSQL, just in case one of the other database types we support does not support `CAST` in the way you have set it up here.

Reply:
@dimensi Also, SQL query commands should always be uppercase: eg: `CAST`.

Reply:
ok